src/Components/ArticleList.jsx

```diff
@@ -1,28 +1,55 @@
-import { useEffect } from "react";
-import { useState } from "react";
+import { useEffect, useState } from "react";
 import ArticleCard from "./ArticleCard";
+import { getArticles } from "../api/api";
+import Error from "./Error";
 
 const ArticleList = () => {
   const [articles, setArticles] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
 
   useEffect(() => {
-    fetch("https://my-nc-news-yyln.onrender.com/api/articles")
-      .then((response) => {
-        return response.json();
-      })
+    getArticles()
       .then((articleData) => {
-        console.log(articleData);
-        setArticles(articleData.articles);
+        setArticles(articleData);
+        setError(null);
+      })
+      .catch((error) => {
+        setError(error);
+      })
+      .then(() => {
+        setLoading(false);
       });
```

Comment:
careful using a `.then` after a `.catch` - if you have code you want to run regardless of whether the `Promise` rejects _or_ resolves successfully as with this `setLoading(false)` you could use `.finally` ([docs here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally))